src/shop/api.py

```diff
@@ -62,20 +64,25 @@ def create_product_transaction(product_id, trx_type, qty, reference=None):
     ct = None
     if reference is not None:
         ct = ContentType.objects.get_for_model(reference)
-    trx_obj = product_obj.transactions.create(
-        trx_type=trx_type,
-        qty=qty,
+    trx_obj = product_obj.transactions.create(trx_type=trx_type, qty=qty)
+    trx_obj.states.create(
+        status=enums.TrxStatus.PENDING,  # It's here to avoid any confusion
         reference_ct=ct,
         reference_id=reference.pk if reference is not None else None
     )
     return trx_obj
 
 
 @transaction.atomic
+def finalize_product_transaction(trx_id):
```

Comment:
There should be a way of specifying a reference both when finalizing and cancelling.